File: app/src/main/java/org/ole/planet/myplanet/ui/dashboard/BellDashboardFragment.java
package org.ole.planet.myplanet.ui.dashboard;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.flexbox.FlexDirection;
import com.google.android.flexbox.FlexboxLayout;

import org.ole.planet.myplanet.R;
import org.ole.planet.myplanet.base.BaseContainerFragment;
import org.ole.planet.myplanet.datamanager.DatabaseService;
import org.ole.planet.myplanet.model.RealmMeetup;
import org.ole.planet.myplanet.model.RealmMyCourse;
import org.ole.planet.myplanet.model.RealmMyLibrary;
import org.ole.planet.myplanet.model.RealmMyTeam;
import org.ole.planet.myplanet.model.RealmSubmission;
import org.ole.planet.myplanet.model.RealmUserModel;
import org.ole.planet.myplanet.service.UserProfileDbHandler;
import org.ole.planet.myplanet.ui.SettingActivity;
import org.ole.planet.myplanet.ui.course.CourseFragment;
import org.ole.planet.myplanet.ui.course.TakeCourseFragment;
import org.ole.planet.myplanet.ui.library.LibraryFragment;
import org.ole.planet.myplanet.ui.mymeetup.MyMeetupDetailFragment;
import org.ole.planet.myplanet.ui.news.NewsFragment;
import org.ole.planet.myplanet.ui.submission.MySubmissionFragment;
import org.ole.planet.myplanet.ui.team.MyTeamsDetailFragment;
import org.ole.planet.myplanet.ui.userprofile.AchievementFragment;
import org.ole.planet.myplanet.ui.userprofile.UserProfileFragment;
import org.ole.planet.myplanet.utilities.Constants;
import org.ole.planet.myplanet.utilities.TimeUtils;
import org.ole.planet.myplanet.utilities.Utilities;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import io.realm.Case;
import io.realm.Realm;
import io.realm.RealmObject;
import io.realm.RealmResults;

/**
 * A placeholder fragment containing a simple view.
 */
public class BellDashboardFragment extends BaseDashboardFragment  {

    public static final String PREFS_NAME = "OLE_PLANET";
    TextView tvCommunityName, tvDate;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_home_bell, container, false);
        declareElements(view);
        onLoaded(view);
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        tvDate.setText(TimeUtils.formatDate(new Date().getTime()));
        tvCommunityName.setText(model.getPlanetCode());
        ((DashboardActivity) getActivity()).getSupportActionBar().hide();
    }

    private void declareElements(View view) {
        tvDate = view.findViewById(R.id.txt_date);
        tvCommunityName = view.findViewById(R.id.txt_community_name);
//        view.findViewById(R.id.iv_setting).setOnClickListener(V->{
//            startActivity(new Intent(getActivity(), SettingActivity.class));
//        });
        initView(view);
        view.findViewById(R.id.ll_achievement).setOnClickListener(v->homeItemClickListener.openCallFragment(new AchievementFragment()));
        view.findViewById(R.id.ll_news).setOnClickListener(v->homeItemClickListener.openCallFragment(new NewsFragment()));
        view.findViewById(R.id.myLibraryImageButton).setOnClickListener(v ->{
            openHelperFragment(new LibraryFragment());

        });
        view.findViewById(R.id.myCoursesImageButton).setOnClickListener(v->{
            openHelperFragment(new CourseFragment());
        });
    }

    private void openHelperFragment(Fragment f)
    {
        Fragment temp = f;
        Bundle b = new Bundle();
        b.putBoolean("isMyCourseLib", true);
        temp.setArguments(b);
        homeItemClickListener.openCallFragment(temp);
        View.OnClickListener showToast = new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(getContext(),"Feature Not Available",Toast.LENGTH_LONG).show();
            }
        };
        view.findViewById(R.id.ll_messages).setOnClickListener(showToast);
        view.findViewById(R.id.ll_calendar).setOnClickListener(showToast);
        view.findViewById(R.id.ll_contacts).setOnClickListener(showToast);
        view.findViewById(R.id.ll_dictionaries).setOnClickListener(showToast);
        view.findViewById(R.id.ll_help_wanted).setOnClickListener(showToast);
    }

}

File: app/src/main/java/org/ole/planet/myplanet/ui/dashboard/BellDashboardFragment.kt
package org.ole.planet.myplanet.ui.dashboard

import android.content.DialogInterface
import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.Fragment
import kotlinx.android.synthetic.main.card_profile_bell.ll_badges
import kotlinx.android.synthetic.main.card_profile_bell.txt_community_name
import kotlinx.android.synthetic.main.card_profile_bell.txt_date
import kotlinx.android.synthetic.main.card_profile_bell.view.fab_feedback
import kotlinx.android.synthetic.main.fragment_home_bell.add_resource
import kotlinx.android.synthetic.main.fragment_home_bell.view.fab_my_activity
import kotlinx.android.synthetic.main.fragment_home_bell.view.fab_my_progress
import kotlinx.android.synthetic.main.fragment_home_bell.view.fab_notification
import kotlinx.android.synthetic.main.fragment_home_bell.view.fab_survey
import kotlinx.android.synthetic.main.home_card_courses.view.myCoursesImageButton
import kotlinx.android.synthetic.main.home_card_library.view.myLibraryImageButton
import kotlinx.android.synthetic.main.home_card_mylife.view.myLifeImageButton
import kotlinx.android.synthetic.main.home_card_teams.view.ll_home_team
import org.ole.planet.myplanet.MainApplication
import org.ole.planet.myplanet.R
import org.ole.planet.myplanet.base.BaseResourceFragment
import org.ole.planet.myplanet.model.RealmCertification
import org.ole.planet.myplanet.model.RealmCourseProgress
import org.ole.planet.myplanet.model.RealmExamQuestion
import org.ole.planet.myplanet.ui.course.CourseFragment
import org.ole.planet.myplanet.ui.course.MyProgressFragment
import org.ole.planet.myplanet.ui.feedback.FeedbackListFragment
import org.ole.planet.myplanet.ui.library.AddResourceFragment
import org.ole.planet.myplanet.ui.library.LibraryFragment
import org.ole.planet.myplanet.ui.mylife.LifeFragment
import org.ole.planet.myplanet.ui.survey.SurveyFragment
import org.ole.planet.myplanet.ui.team.TeamFragment
import org.ole.planet.myplanet.utilities.TimeUtils
import java.util.Date

class BellDashboardFragment : BaseDashboardFragment() {
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_home_bell, container, false)
        declareElements(view)
        onLoaded(view)
        return view
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        txt_date.text = TimeUtils.formatDate(Date().time)
        txt_community_name.text = model.planetCode
        (activity as DashboardActivity?)?.supportActionBar?.hide()
        add_resource.setOnClickListener { v: View? ->
            AddResourceFragment().show(
                childFragmentManager, getString(R.string.add_res)
            )
        }
        showBadges()
        if (!model.id.startsWith("guest") && TextUtils.isEmpty(model.key) && MainApplication.showHealthDialog) {
            AlertDialog.Builder(activity!!)
                .setMessage(getString(R.string.health_record_not_available_sync_health_data))
                .setPositiveButton(getString(R.string.sync)) { dialogInterface: DialogInterface?, i: Int ->
                    syncKeyId()
                    MainApplication.showHealthDialog = false
                }.setNegativeButton(getString(R.string.cancel), null).show()
        }
    }

    private fun showBadges() {
        ll_badges.removeAllViews()
        val list = RealmCourseProgress.getPassedCourses(
            mRealm, BaseResourceFragment.settings.getString("userId", "")
        )
        for (sub in list) {
            val star =
                LayoutInflater.from(activity).inflate(R.layout.image_start, null) as ImageView
            val examId = if (sub.parentId.contains("@")) sub.parentId.split("@")
                .toTypedArray()[0] else sub.parentId
            val courseId =
                if (sub.parentId.contains("@")) sub.parentId.split("@").toTypedArray()[1] else ""
            val questions =
                mRealm.where(RealmExamQuestion::class.java).equalTo("examId", examId).count()
            setColor(questions, courseId, star)
            ll_badges.addView(star)
        }
    }

    private fun setColor(questions: Long, courseId: String, star: ImageView) =
        if (RealmCertification.isCourseCertified(mRealm, courseId)) {
            star.setColorFilter(resources.getColor(R.color.colorPrimary))
        } else {
            star.setColorFilter(resources.getColor(R.color.md_blue_grey_300))
        }

    private fun declareElements(view: View) {
        initView(view)
        view.ll_home_team.setOnClickListener { homeItemClickListener.openCallFragment(TeamFragment()) }
        view.myLibraryImageButton.setOnClickListener { openHelperFragment(LibraryFragment()) }
        view.myCoursesImageButton.setOnClickListener { openHelperFragment(CourseFragment()) }
        view.fab_my_progress.setOnClickListener { openHelperFragment(MyProgressFragment()) }
        view.fab_my_activity.setOnClickListener { openHelperFragment(MyActivityFragment()) }
        view.fab_survey.setOnClickListener { openHelperFragment(SurveyFragment()) }
        view.fab_feedback.setOnClickListener { openHelperFragment(FeedbackListFragment()) }
        view.myLifeImageButton.setOnClickListener {
            homeItemClickListener.openCallFragment(
                LifeFragment()
            )
        }
        view.fab_notification.setOnClickListener { showNotificationFragment() }
    }

    private fun openHelperFragment(f: Fragment) {
        val b = Bundle()
        b.putBoolean("isMyCourseLib", true)
        f.arguments = b
        homeItemClickListener.openCallFragment(f)
    }

    companion object {
        const val PREFS_NAME = "OLE_PLANET"
    }
}